File: 99-Continue/145/TicketBookingSystem.java
import java.util.*;
import java.util.concurrent.ArrayBlockingQueue;

public class TicketBookingSystem {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        Queue<String> queue = new ArrayBlockingQueue<>(size);
        
        int n= sc.nextInt();
        sc.nextLine();
        for(int i=0;i<n;i++){
            String s = sc.nextLine();
            
            switch(s.split(" ")[0]){
                case "ENQUEUE" -> {
                    if (!queue.offer(s.split(" ")[1])) {
                        System.out.println("Queue is full. Cannot add " + s.split(" ")[1]);
                    }
                }
                case "DISPLAY" -> {
                    if(queue.isEmpty()){
                        System.out.println("Queue is empty.");
                    }
                    else{
                        System.out.print("Current Queue: ");
                        for(String q:queue){
                            System.out.print(q+" ");
                        }
                        System.out.println();
                    }
                }
                case "DEQUEUE" -> {
                    if(queue.isEmpty()){
                        System.out.println("Served Customer: Queue is empty. No customer to serve.");
                    }
                    else{
                        System.out.println("Served Customer: "+queue.poll());
                    }
                }
            }
        }
        sc.close();
    }
}

File: 99-Continue/145/TicketBookingSystem.py
from collections import deque

def main():
    queue = deque()

    size = int(input())
    n = int(input())

    for _ in range(n):

        s = input().split()

        match s[0]:

            case "ENQUEUE":
                if len(queue) < size:
                    queue.append(s[1])
                else:
                    print(f"Queue is full. Cannot add {s[1]}")

            case "DISPLAY":
                disp = ""
                if not queue:
                    print("Queue is empty.")
                else:
                    print("Current Queue: ",end="")
                    for s in queue:
                        disp=disp+s+" "
                    print(disp)

            case "DEQUEUE":
                if not queue:
                    print("Served Customer: Queue is empty. No customer to serve.")
                else:
                    customer = queue.popleft()
                    print(f"Served Customer: {customer}")

if __name__ == "__main__":
    main()